File: 53. Maximum Subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        //if (nums.size() == 1) return nums[0];
        int res = INT_MIN, cur = 0;
        for (int i = 0; i < nums.size(); i++) {
            cur += nums[i];
            res = max(res, cur);
            if (cur < 0) {
                cur = 0;
            }
        }
        return res;
    }
};

File: java/53. Maximum Subarray.java
// O(n)
class Solution {
    public int maxSubArray(int[] nums) {
        int res = nums[0], max = nums[0];
        for (int i = 1; i < nums.length; ++i) {
            if (max > 0) {
                max += nums[i];
            } else {
                max = nums[i];
            }
            res = Math.max(res, max);
        }
        return res;
    }
}
